### DISCUSSION TITLE
[WASM] gRPC stream limits

### DISCUSSION THREAD

**Discussion by limeniye** [unoplatform/uno#12450] [Q&A]
I need to connect several streams to a gRPC server. But instead, only a few connect.
It looks like I have some sort of stream limits.

![зображення](https://github.com/unoplatform/uno/assets/69314237/e111fba5-1ceb-48a8-a02a-e80ba918d325)

This problem does not exist on the Windows platform.

Perhaps I am not aware of some nuances.

Maybe someone has an idea what the problem might be.

**Comment by carldebilly** [comment]
There is no official guidance (yet) from Uno to integrate gRPC into a wasm application.

Which "flavor" of gRPC are you using right now? If it's gRPC-web (http 1.1), you could be reaching the limit of the number of connections the browser can make to the same hostname. A work-around would be to switch to http2 or to create CNAMEs to have different host names.

**Reply by limeniye** [reply]
@carldebilly, thank you for your answer.

I'll attach option information:

[Startup.cs](https://github.com/ligric/ligric/blob/main/src/services/Ligric.Service.CryptoApisService/Ligric.Service.CryptoApisService.Api/Startup.cs#L111)

```C#
endpoints.MapGrpcService<Services.UserApisService>().RequireCors(CORS_POLICY).EnableGrpcWeb();
endpoints.MapGrpcService<Services.FuturesService>().RequireCors(CORS_POLICY).EnableGrpcWeb()
```


In general I'm using Envoy proxy gateway for transcoding http1 to http2:
https://user-images.githubusercontent.com/69314237/231535897-68728e85-e25d-4282-b118-d4800256f403.svg

But also I have been tried it with only started server app without Envoy.
The both cases give me the same result. 

```
A work-around would be to switch to http2 or to create CNAMEs to have different host names.
```
Hm. But with HTTP2 WASM doesn't work at all.

**Reply by limeniye** [reply]
@carldebilly, I found this title
https://docs.diffusiondata.com/cloud/latest/manual/html/designguide/solution/support/connection_limitations.html

Maybe this is my problem 🤔

Updated:
Ok, if I understood it well, I cant use more then 6 streams for Chrome browser because Chrome are support limited http1 connections.
Consequently I need to use the HTTP 2 for multiplexing connections. But WASM is not support HTTP/2 or HTTP/3.

So what can I do?
